src/pages/workspaces/workspace/tools/LaunchAnalysisModal.js

```diff
@@ -0,0 +1,197 @@
+            onChange: e => this.setState({ filterText: e.target.value, pageNumber: 1 })
+          }
+        })
+      ],
+      showX: true,
+      width: 'calc(100% - 2rem)',
+      okButton: buttonPrimary({
+        onClick: () => this.launch(),
+        disabled: launching
+      }, [launching ? 'Launching...' : 'Launch'])
+    }, [
+      Utils.cond(
+        [attributeNames && entities, () => this.renderMain()],
+        [attributeFailure || entityFailure, () => this.renderError()],
+        () => spinner()
+      )
+    ])
+  }
+
+  componentDidMount() {
+    const { workspaceId: { namespace, name }, config: { rootEntityType } } = this.props
+
+    Rawls.workspace(namespace, name).entities().then(
+      entities => {
+        const { attributeNames, idName } = entities[rootEntityType]
+        this.setState({ attributeNames, idName })
+      },
+      attributeFailure => this.setState({ attributeFailure })
+    )
+
+    Rawls.workspace(namespace, name).entity(rootEntityType).then(
+      entities => this.setState({ entities }),
+      entityFailure => this.setState({ entityFailure })
+    )
+  }
+
+  renderMain = () => {
+    const { itemsPerPage, pageNumber, entities, filterText, launchError } = this.state
+    const filteredEntities = _.filter(entity => entity.name.includes(filterText), entities)
+
+    return h(Fragment, [
+      div({ style: { overflowX: 'auto', margin: '0 -1.25rem', padding: '0 1.25rem' } }, [
+        div({ style: { display: 'table', marginBottom: '0.5rem' } }, [
+          h(ScrollWithHeader, {
+            header: this.renderTableHeader(),
+            negativeMargin: '1.25rem',
+            children: [this.renderTableBody(filteredEntities)]
+          })
+        ])
+      ]),
+      div({ style: { marginTop: 10 } }, [
+        paginator({
+          filteredDataLength: filteredEntities.length,
+          setPageNumber: pageNumber => this.setState({ pageNumber }),
+          pageNumber,
+          setItemsPerPage: itemsPerPage => this.setState({ itemsPerPage }),
+          itemsPerPage
+        })
+      ]),
+      div({ style: { marginTop: 10, textAlign: 'right', color: Style.colors.error } }, [launchError])
+    ])
+  }
+
+  renderError = () => {
+    const { attributeFailure, entityFailure } = this.state
+
+    return div({}, [
+      div({}, 'Unable to load data entities'),
+      attributeFailure && div({}, attributeFailure),
+      entityFailure && div({}, entityFailure)
+    ])
+  }
+
+  renderTableHeader = () => {
+    const { attributeNames, idName } = this.state
+
+    const cellStyle = {
+      overflow: 'hidden', textOverflow: 'ellipsis',
+      fontWeight: 500, fontSize: 12, padding: '0.5rem 19px'
+    }
+
+    /*
+     * FIXME: width: 0 solves an issue where this header sometimes takes more room than
+     * it needs and messes up the layout of the entire table. Related to the display: table
+     * that's used to make style apply beyond the viewport of a scrolling component
```

Comment:
Is it worth throwing in a bug so we can approach this separately and with fresh eyes?

Reply:
Will file one.

Reply:
Filed #205